tests/test_sendinblue_backend.py

```diff
@@ -0,0 +1,444 @@
+        self.message.to = ['alice@example.com', 'Bob <bob@example.com>', 'celia@example.com']
+        self.message.cc = ['cc@example.com']
+        self.message.body = "Hi :name. Welcome to :group at :site."
+        self.message.merge_data = {
+            'alice@example.com': {':name': "Alice", ':group': "Developers"},
+            'bob@example.com': {':name': "Bob"},  # and leave :group undefined
+        }
+        self.message.merge_global_data = {
+            ':group': "Users",
+            ':site': "ExampleCo",
+        }
+
+        with self.assertRaises(AnymailUnsupportedFeature):
+            self.message.send()
+
+    def test_default_omits_options(self):
+        """Make sure by default we don't send any ESP-specific options.
+
+        Options not specified by the caller should be omitted entirely from
+        the API call (*not* sent as False or empty). This ensures
+        that your ESP account settings apply by default.
+        """
+        self.message.send()
+        data = self.get_api_call_json()
+        self.assertNotIn('attachment', data)
+        self.assertNotIn('tag', data)
+        self.assertNotIn('headers', data)
+        self.assertNotIn('send_at', data)
+
+    def test_esp_extra(self):
+        self.message.tags = ["tag"]
+        # SendinBlue doesn't offer any esp-extra but we will test
+        # with some extra of SendGrid to see if it's work in the future
+        self.message.esp_extra = {
+            'ip_pool_name': "transactional",
+            'asm': {  # subscription management
+                'group_id': 1,
+            },
+            'tracking_settings': {
+                'subscription_tracking': {
+                        'enable': True,
+                        'substitution_tag': '[unsubscribe_url]',
+                },
+            },
+        }
+        self.message.send()
+        data = self.get_api_call_json()
+        # merged from esp_extra:
+        self.assertEqual(data['ip_pool_name'], "transactional")
+        self.assertEqual(data['asm'], {'group_id': 1})
+        self.assertEqual(data['tracking_settings']['subscription_tracking'],
+                         {'enable': True, 'substitution_tag': "[unsubscribe_url]"})
+        # make sure we didn't overwrite Anymail message options:
+        self.assertEqual(data['headers']["X-Mailin-tag"], "tag")
+
+    # noinspection PyUnresolvedReferences
+    def test_send_attaches_anymail_status(self):
+        """ The anymail_status should be attached to the message when it is sent """
+        # the DEFAULT_RAW_RESPONSE above is the *only* success response SendinBlue returns,
+        # so no need to override it here
+        msg = mail.EmailMessage('Subject', 'Message', 'from@example.com', ['to1@example.com'],)
+        sent = msg.send()
+        self.assertEqual(sent, 1)
+        self.assertEqual(msg.anymail_status.status, {'queued'})
+        self.assertEqual(msg.anymail_status.recipients['to1@example.com'].status, 'queued')
```

Comment:
This should also check that the `msg.anymail_status.message_id` and `msg.anymail_status.recipients['to1@example.com'].message_id` get set to the message_id from SiB's API response.

Reply:
Done